src/main/java/org/wso2/extension/siddhi/window/uniquelengthbatch/UniqueLengthBatchWindowProcessor.java

```diff
@@ -0,0 +1,283 @@
+import org.wso2.siddhi.core.event.ComplexEventChunk;
+import org.wso2.siddhi.core.event.state.StateEvent;
+import org.wso2.siddhi.core.event.stream.StreamEvent;
+import org.wso2.siddhi.core.event.stream.StreamEventCloner;
+import org.wso2.siddhi.core.executor.ConstantExpressionExecutor;
+import org.wso2.siddhi.core.executor.ExpressionExecutor;
+import org.wso2.siddhi.core.executor.VariableExpressionExecutor;
+import org.wso2.siddhi.core.query.processor.Processor;
+import org.wso2.siddhi.core.query.processor.stream.window.FindableProcessor;
+import org.wso2.siddhi.core.query.processor.stream.window.WindowProcessor;
+import org.wso2.siddhi.core.table.EventTable;
+import org.wso2.siddhi.core.util.collection.operator.Finder;
+import org.wso2.siddhi.core.util.collection.operator.MatchingMetaStateHolder;
+import org.wso2.siddhi.core.util.parser.OperatorParser;
+import org.wso2.siddhi.query.api.definition.Attribute;
+import org.wso2.siddhi.query.api.exception.ExecutionPlanValidationException;
+import org.wso2.siddhi.query.api.expression.Expression;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/*
+ * UniqueLengthBatch Window
+ * Sample Query:
+ * from inputStream#window.unique:lengthBatch(attribute1,3,true)
+ * select attribute1, attribute2
+ * insert into outputStream;
+ *
+ * Description:
+ * In the example query given, 3 is the length of the window and attribute1 is the unique attribute.
+ * According to the given attribute it will give length number of unique events .
+ * attribute true is to keep first unique default value is last unique.
+ *
+ * @since 1.0.0
+ */
+public class UniqueLengthBatchWindowProcessor extends WindowProcessor implements FindableProcessor {
+
+    private int windowLength;
+    private int count = 0;
+    private ComplexEventChunk<StreamEvent> currentEventChunk = new ComplexEventChunk<StreamEvent>(false);
+    private ComplexEventChunk<StreamEvent> eventsToBeExpired = null;
+    private ExecutionPlanContext executionPlanContext;
+    private StreamEvent resetEvent = null;
+    private VariableExpressionExecutor uniqueKey;
+    private Map<Object, StreamEvent> uniqueEventMap = new HashMap<>();
+    private boolean isFirstUniqueEnabled = false;
+
+    /**
+     * The init method of the WindowProcessor, this method will be called before other methods.
+     *
+     * @param attributeExpressionExecutors the executors of each function parameters
+     * @param executionPlanContext         the context of the execution plan
+     */
+    @Override
+    protected void init(ExpressionExecutor[] attributeExpressionExecutors, ExecutionPlanContext executionPlanContext) {
+        this.executionPlanContext = executionPlanContext;
+        this.eventsToBeExpired = new ComplexEventChunk<>(false);
+        if (attributeExpressionExecutors.length == 2) {
+            this.uniqueKey = (VariableExpressionExecutor) attributeExpressionExecutors[0];
```

Comment:
We need to check whether `attributeExpressionExecutors[0] instanceof VariableExpressionExecutor`